### PR TITLE
Reduce Citus default shard count to 16

### LINKED ISSUE #5622: Citus load balance queries across all nodes
### Problem

Since citus release v11.0, it's possible to query from any node (coordinator or worker). For workload which requires a lot of client connections or the coordinator node becomes the resource bottleneck, spreading the queries should achieve higher utilization of the cluster resource thus increase query throughput.

### Solution

Spread the queries across all nodes by creating a k8s service or utilizing postgresql jdbc driver side load balancing.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

* Document recommended Citus cluster configuration
* Reduce Citus default shard count to 16
* Update Citus coordinator config to allow replication

**Related issue(s)**:

Fixes #5622 

**Notes for reviewer**:

Per the discussion in scrum, the PR reduces the Citus default shard count to 16. The test result with 3 coordinators can be found in #5622.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by edwin-greene — DISMISSED** [review]

**Comment by xin-hedera** [comment]
Need to add additional doc and chart config changes, will re-open for review

File: docs/database.md [hunk 1]
```diff
@@ -199,15 +199,26 @@ transactions in the balance and record streams. These issues should only appear
...
 7. Stop the [Importer](/docs/importer/README.md) process.
-8. Update the Importer to point to the new citus DB and start it.
+8. Update the Importer to point to the new Citus DB and start it.
+
+The recommended configuration for a production Citus cluster. Note the storage size depends on the amount of data to
+ingest and keep.
 
+| Node Type   | Count | vCPU | Memory | Disk Storage |
+|-------------|-------|------|--------|--------------|
+| Coordinator | 1     | 4    | 26 GB  | 64 GB        |
```
Comment (steven-sheehy) [inline docs/database.md:223]: I thought we tested 3 coordinators as the recommended number?

Comment (steven-sheehy) [inline docs/database.md:223]: Maybe move this under step 1 which mentions creating cluster with enough resources.

File: docs/database.md [hunk 2]
```diff
@@ -199,15 +199,26 @@ transactions in the balance and record streams. These issues should only appear
...
+| Worker      | 3     | 4    | 26 GB  | 512 GB       |
```
Comment (steven-sheehy) [inline docs/database.md:224]: We should size disks based upon mainnet with all historical data.

File: docs/database.md (hunk 1 above)
Comment (xin-hedera) [reply docs/database.md:223]: sure, will increase it to 3 and keep the vCPU / memory, we can fine tune the coordinator resources later if lower resources per coordinator provide good compromise between cost and performance

Comment (xin-hedera) [reply docs/database.md]: moved to the beginning of the section

File: docs/database.md (hunk 2 above)
Comment (xin-hedera) [reply docs/database.md]: done

**Review by steven-sheehy — APPROVED** [review]

**Review by mgoelswirlds — APPROVED** [review]

**Pull request merged**